packages/tangle/booker.go

```diff
@@ -171,19 +202,35 @@ func (b *Booker) Shutdown() {
 	b.MarkersManager.Shutdown()
 }
 
-// parentsBranchIDs returns the BranchIDs of a Message's parents.
-func (b *Booker) parentsBranchIDs(message *Message) (branchIDs ledgerstate.BranchIDs) {
-	branchIDs = make(ledgerstate.BranchIDs)
+// allMessagesContainTransactions checks all passed messages contain a transaction
+func (b *Booker) allMessagesContainTransactions(messageIDs MessageIDs) (areAllTransactions bool) {
+	areAllTransactions = true
+	for _, messageID := range messageIDs {
+		if !areAllTransactions {
```

Comment:
If you put this `if` after the `Consume` you break out of the loop immediately on a non tx containing msg, instead of iterating once more for no reason.